src/agent/src/rpc.rs

```diff
@@ -1737,6 +1774,137 @@ fn load_kernel_module(module: &protocols::agent::KernelModule) -> Result<()> {
     }
 }
 
+fn pull_image_from_registry(
+    image: &str,
+    cid: &str,
+    source_creds: &Option<&str>,
+    use_aa: bool,
+    agent_started: bool,
+) -> Result<()> {
+    let source_image = format!("{}{}", "docker://", image);
+
+    let manifest_path = format!("/tmp/{}/image_manifest", cid);
+    let target_path_manifest = format!("dir://{}", manifest_path);
+
+    // Define the target transport and path for the OCI image, without signature
+    let oci_path = format!("/tmp/{}/image_oci:latest", cid);
+    let target_path_oci = format!("oci://{}", oci_path);
+
+    fs::create_dir_all(&manifest_path)?;
+    fs::create_dir_all(&oci_path)?;
+
+    info!(sl!(), "Attempting to pull image {}...", &source_image);
+
+    let mut pull_command = Command::new(SKOPEO_PATH);
+    pull_command
+        .arg("--insecure-policy")
+        .arg("copy")
+        .arg(source_image)
+        .arg(&target_path_manifest);
+
+    if let Some(source_creds) = source_creds {
+        pull_command.arg("--src-creds").arg(source_creds);
+    }
+
+    if use_aa {
+        if !agent_started {
+            init_attestation_agent();
+        }
+
+        // not 100% sure we actually need this. the arg gets ignored anyway
+        pull_command
+            .arg("--decryption-key")
+            .arg("provider:attestation-agent:offline_fs_kbc::null")
+            .env("OCICRYPT_KEYPROVIDER_CONFIG", OCICRYPT_CONFIG_PATH);
+
+        //TODO: check if image is really encrypted since skopeo doesn't
+        //      seem to care
```

Comment:
Needs a GitHub issue URL for tracking this remaining item.

Reply:
I've been thinking about this a little bit, and it probably isn't worth it to add anything to Kata here. Instead, I will open an issue on the new `image-rs` project to make sure that copying an image will fail if you provide a keyprovider but the image is not encrypted.